invenio_users_resources/records/mappings/os-v1/users/user-v2.0.0.json

```diff
@@ -1,4 +1,25 @@
 {
+  "settings": {
+    "index": {
+      "max_ngram_diff": 2
+    },
+    "analysis": {
+      "analyzer": {
+        "ngram_3_to_5_analyzer": {
+          "type": "custom",
+          "tokenizer": "ngram_tokenizer_3_to_5",
+          "filter": ["lowercase"]
+        }
+      },
+      "tokenizer": {
+        "ngram_tokenizer_3_to_5": {
+          "type": "ngram",
+          "min_gram": 3,
+          "max_gram": 5
+        }
+      }
+    }
+  },
```

Comment:
This looks like, is doing a good job to allow to find partial matches, however we would need to do some performance tests to ensure that it doesn't affect negatively too much.

Reply:
Have you tested that with 2 letter names e.g `Lu`? I have the feeling that the `min_gram` should be 2...

Reply:
We were only giving results when typing 3 letters if I am not mistaken, but min_gram and max_gram can be adjusted and tested much more, specially to take into consideration how much will the index grow and how does it affect to performance, lowering the ngrams to 2 will end up doing way more matches but it should be tested and maybe it's worth to do it :+1:

Reply:
Have you considered also this https://www.elastic.co/guide/en/elasticsearch/reference/current/search-as-you-type.html to add in the specific field we need instead of ngram?

Reply:
search-as-you-type is for the suggestion API and so we can't use it here. However we can use [edge n gram matching](https://opensearch.org/docs/latest/search-plugins/searching-data/autocomplete/#edge-n-gram-matching) instead, which I am using in this PR